src/CommandingOfficers/DefendPeace/RoseThorn/Tech.java

```diff
@@ -570,23 +570,27 @@ private static UnitModel createBattleMechModel(GameReadyModels grms)
     UnitModel mdTank = grms.unitModels.getAllModels(UnitModel.LAND | UnitModel.ASSAULT, matchOnAny).get(1);
     UnitModel antiAir = grms.unitModels.getUnitModel(UnitModel.LAND | UnitModel.SURFACE_TO_AIR, matchOnAny);
 
-    UnitModel BattleMech = mdTank.clone();
-    BattleMech.name = BATTLEMECH_NAME;
-    BattleMech.role = BattleMech.role | UnitModel.SURFACE_TO_AIR;
-    BattleMech.costBase = mdTank.costBase*2 + (antiAir.costBase/2);
-    BattleMech.abilityPowerValue = 20;
-    BattleMech.maxFuel = 30;
-    BattleMech.maxAmmo = 10;
-    BattleMech.visionRange = 2;
-    BattleMech.baseMovePower = 4;
-    BattleMech.baseMoveType = new FootMech();
-    BattleMech.healableHabs = new HashSet<TerrainType>(); // BattleMechs have specialized parts, not easy to repair.
+    var BattleMech = mdTank.toBuilder();
+    BattleMech.name(BATTLEMECH_NAME);
+    BattleMech.costBase(mdTank.costBase*2 + (antiAir.costBase/2));
+    BattleMech.abilityPowerValue(20);
+    BattleMech.maxFuel(30);
+    BattleMech.maxAmmo(10);
+    BattleMech.visionRange(2);
+    BattleMech.baseMovePower(4);
+    BattleMech.baseMoveType(new FootMech());
```

Comment:
This doesn't set `role` or `healableHabs`. I don't think `setCalculatedProps` handles the `SURFACE_TO_AIR` property, either.
Guess I need to read up on lombok builders, too.